### DISCUSSION TITLE
Example of piping through vd

### DISCUSSION THREAD

**Discussion by ivandatasci** [saulpw/visidata#1818] [General]
How do I pipe through visidata?

For example

myfile.csv

sensor,  temperature
1, 25.0
2, 30.1
5, 29.0
10, 19.0
20, 40.5

Now I want to
1) define temperature as a float type
2) filter the rows so that only temperatures above 20.0 are retained
3) define sensor as an integer type
4) sort in decreasing order of sensor number
5) output to tsv

So, this would be the scaffold:

cat myfile.csv | vd -f csv [some commands here] -o /dev/stdout > myfile.filtered.tsv

What commands should I pass to vd to perform tasks 1 through 5 non-interactively?

Thank you.

Ivan

**Comment by saulpw** [comment]
Do the commands manually, use Shift+D to open the cmdlog, save it as a .vdj file, then `-b -p foo.vdj`.
